src/main/java/edu/ie3/datamodel/io/source/file/FileDataSource.java

```diff
@@ -0,0 +1,141 @@
+              FileType fileType;
+              try {
+                fileType = FileType.getFileType(fileName);
+              } catch (ParsingException e) {
+                log.warn("Unable to load load profile meta data for {}", fileName, e);
+                fileType = null; // will be filtered out
+              }
+              /* Extract meta information from file path and enhance it with the file path itself */
+              IndividualTimeSeriesMetaInformation metaInformation =
+                  namingStrategy.individualTimeSeriesMetaInformation(filePath.toString());
+              return new FileIndividualTimeSeriesMetaInformation(
+                  metaInformation,
+                  Path.of(FileNamingStrategy.removeFileNameEnding(fileName)),
+                  fileType);
+            })
+        .filter(meta -> meta.getFileType() != null)
+        .filter(
+            metaInformation ->
+                columnSchemes == null
+                    || columnSchemes.length == 0
+                    || Stream.of(columnSchemes)
+                        .anyMatch(scheme -> scheme.equals(metaInformation.getColumnScheme())));
+  }
+
+  /**
+   * Receive the information for specific load profile time series. They are given back mapped to
+   * their uuid.
+   *
+   * @return A mapping from profile to the load profile time series meta information
+   */
+  public Stream<FileLoadProfileMetaInformation> getLoadProfileMetaInformation(
+      LoadProfile... profiles) {
+    FileNamingStrategy namingStrategy = getNamingStrategy();
+
+    return getTimeSeriesFilePaths(namingStrategy.getLoadProfileTimeSeriesPattern()).parallelStream()
+        .map(
+            filePath -> {
+              String fileName = filePath.getFileName().toString();
+              FileType fileType;
+              try {
+                fileType = FileType.getFileType(fileName);
+              } catch (ParsingException e) {
+                log.warn("Unable to load load profile meta data for {}", fileName, e);
+                fileType = null; // will be filtered out
+              }
+
+              /* Extract meta information from file path and enhance it with the file path itself */
+              LoadProfileMetaInformation metaInformation =
+                  namingStrategy.loadProfileTimeSeriesMetaInformation(filePath.toString());
+              return new FileLoadProfileMetaInformation(
+                  metaInformation.getProfile(),
+                  Path.of(FileNamingStrategy.removeFileNameEnding(fileName)),
+                  fileType);
+            })
+        .filter(meta -> meta.getFileType() != null)
+        .filter(
+            metaInformation ->
+                profiles == null
+                    || profiles.length == 0
+                    || Stream.of(profiles)
+                        .anyMatch(
+                            profile -> profile.getKey().equals(metaInformation.getProfile())));
+  }
+}
```

Comment:
This is pretty much the same code as above in `getIndividualTimeSeriesMetaInformation`. Maybe you can find a way to abstract the common functionality.